client/blocks/conversation-follow-button/button.jsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * @format
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React from 'react';
+import { noop } from 'lodash';
+import { localize } from 'i18n-calypso';
+import Gridicon from 'gridicons';
+
+class ConversationFollowButton extends React.Component {
+	static propTypes = {
+		isFollowing: PropTypes.bool.isRequired,
+		onFollowToggle: PropTypes.func,
+		tagName: PropTypes.oneOfType( [ PropTypes.string, PropTypes.func ] ),
+	};
+
+	static defaultProps = {
+		isFollowing: false,
+		onFollowToggle: noop,
+		tagName: 'button',
+	};
+
+	toggleFollow = event => {
+		if ( event ) {
+			event.preventDefault();
+		}
+
+		if ( this.props.onFollowToggle ) {
+			this.props.onFollowToggle( ! this.props.isFollowing );
+		}
+	};
+
+	render() {
+		const { isFollowing, translate } = this.props;
+		const buttonClasses = [
+			'button',
+			'has-icon',
+			'conversation-follow-button',
+			this.props.className,
+		];
+		const iconSize = 20;
+		const label = isFollowing
+			? translate( 'Following Conversation' )
+			: translate( 'Follow Conversation' );
+
+		if ( this.props.isFollowing ) {
+			buttonClasses.push( 'is-following' );
+		}
+
+		const followingIcon = (
+				<Gridicon key="following" icon="reader-following-conversation" size={ iconSize } />
+			),
+			followIcon = <Gridicon key="follow" icon="reader-follow-conversation" size={ iconSize } />,
```

Comment:
do you mind if we switch these one semicolor/const per line? 
I find comma separated declarations frustrating especially every since `let/const` came out.